woocommerce/navigation
# Migrate existing POC work from WCA

Fixes #10 

Migrates the existing navigation work from WooCommerce Admin to get things started.  Note there are a few things that were required to get this up and running that we may want to update going forward:

* Settings are accessed directly via the `wcNavigation` global on the `window`.  This might be better kept in a store or some client-side interface.
* Directly imported variables and colors in scss instead of prepending through webpack.
* Uses a class component instead of a functional component for render.

Most of this is a direct migration from the existing work so my preference is to iterate on this in follow-up PRs, but I did make some quick decisions around separation of logic so we're not throwing everything into a giant class.  Open to feedback on naming or other items we see necessary here before merging in.

### Screenshots
<img width="544" alt="Screen Shot 2020-07-02 at 2 49 17 PM" src="https://user-images.githubusercontent.com/10561050/86355819-f2439e00-bc73-11ea-8726-d21f18f02202.png">


### Testing

1. Go to one of the registered WC screens (e.g., `/wp-admin/admin.php?page=wc-settings`)
1. Note the new menu is shown (with very little styling).
1. Navigate to a non-WC page.
1. Note the old menu is shown and migrated menu items have been removed.
